a-star_test.go

```diff
@@ -116,3 +119,143 @@ func ExampleShortestPathWithHeuristic() {
 	// Output:
 	// 9 8 6 5 4 3 1
 }
+
+func TestCreateGrid_100X100Nodes(t *testing.T) {
+	expectedNodesConnections := []struct {
+		connectedNode string
+		expectedConnections []string
+	} {
+		{
+			connectedNode: "1",
+			expectedConnections: []string{
+				"2", // right
+				"102", // right-down
+				"101", // down
+				"200", // left-down
+				"100", // left
+				"10000", // left-up
+				"9901", // up
+				"9902", // up-right
+			},
+		},
+		{
+			connectedNode: "245",
+			expectedConnections: []string{
+				"246", // right
+				"346", // right-down
+				"345", // down
+				"344", // left-down
+				"244", // left
+				"144", // left-up
+				"145", // up
+				"146", // up-right
+			},
+		},
+		{
+			connectedNode: "10000",
+			expectedConnections: []string{
+				"9901", // right
+				"1", // right-down
+				"100", // down
+				"99", // left-down
+				"9999", // left
+				"9899", // left-up
+				"9900", // up
+				"9901", // up-right
+			},
+		},
+	}
+
+	g := createGrid(100, 100)
+	for _, e := range expectedNodesConnections {
+		for _, expectedConn := range e.expectedConnections {
+			connExist, _ := g.Adjacent(e.connectedNode, expectedConn)
+			if !connExist {
+				t.Fatalf("Expected grid to have a connection between node %s and node %s", e.connectedNode, expectedConn)
+			}
+		}
+	}
+}
+
+func createGrid(rows, columns int) *Graph {
+	g := New()
+
+	// add nodes
+	totalNumberOfNodes := rows*columns
+	for i := 1; i <= totalNumberOfNodes; i++ {
+		g.Add(strconv.Itoa(i))
+	}
+
+	// connect each node to 8 surrounding nodes
+	for i := 1; i <= totalNumberOfNodes; i++ {
+		nodeKey := strconv.Itoa(i)
+
+		right := getRight(i, columns)
+		left := getLeft(i, columns)
+		down := getDown(i, rows, columns)
+
+		rightDown := getDown(right, rows, columns)
+		downLeft := getDown(left, rows, columns)
+
+		g.Connect(nodeKey, strconv.Itoa(right), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(rightDown), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(down), rand.Intn(5))
+		g.Connect(nodeKey, strconv.Itoa(downLeft), rand.Intn(5))
+	}
+
+
+	return g
+}
+
+func getRight(nodePos, columns int) int {
+	if nodePos% columns == 0 {
+		return nodePos - (columns - 1)
+	}
+	return nodePos + 1
+}
+
+func getLeft(nodePos, columns int) int {
+	if nodePos% columns == 1 {
+		return nodePos + (columns - 1)
+	}
+	return nodePos - 1
+}
+
+func getDown(nodePos, rows, columns int) int {
+	FirstNodeOfLastRow := rows*columns - columns + 1
+	if nodePos >= FirstNodeOfLastRow {
+		return nodePos - (rows*columns - columns)
+	}
+	return nodePos + columns
+}
+
+func BenchmarkShortestPathWithHeuristic_100X100GridOfNodes(b *testing.B) {
+	rows, columns := 100, 100
+	grid := createGrid(rows, columns)
+	randomNodes := createLongListOfRandomKeyNodes(rows * columns, 2000)
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		_, ok := grid.ShortestPathWithHeuristic(randomNodes[i].fromNode, randomNodes[i].toNode, h)
```

Comment:
I see you are using my naive heuristic from the top, which means it will give you `0` as estimated distance for all cases where both nodes "positions" are >10. I'm pretty sure that slows down these benchmarks. Do you not have a heuristic for your use case?